app/src/main/java/org/wikipedia/talk/TalkTopicsActivity.kt

```diff
@@ -99,7 +99,18 @@ class TalkTopicsActivity : BaseActivity() {
                 val pos = data.getIntExtra(WikipediaLanguagesFragment.ACTIVITY_RESULT_LANG_POSITION_DATA, 0)
                 if (pos < WikipediaApp.getInstance().language().appLanguageCodes.size) {
                     funnel.logChangeLanguage()
-                    pageTitle = PageTitle(pageTitle.namespace, StringUtil.removeNamespace(pageTitle.prefixedText),
+
+                    val newNamespace = when {
+                        pageTitle.namespace() == Namespace.USER -> {
+                            UserAliasData.valueFor(WikipediaApp.getInstance().language().appLanguageCodes[pos])
+                        }
+                        pageTitle.namespace() == Namespace.USER_TALK -> {
+                            UserTalkAliasData.valueFor(WikipediaApp.getInstance().language().appLanguageCodes[pos])
+                        }
+                        else -> pageTitle.namespace
+                    }
+
+                    pageTitle = PageTitle(newNamespace, StringUtil.removeNamespace(pageTitle.prefixedText),
```

Comment:
Would it be possible to set `getDisplayText` here? Although this supposed to happen on the user talk page and it would be great to keep consistency.

Reply:
No, I don't see how this can be done without overwriting the language-specific namespace name.  To do this properly we'll need to do a major refactor of how we handle the namespace+displayText internally in PageTitle.

Reply:
Makes sense to me, thanks.